platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
I think functions that work on FileHandle and are fundamental to its operation can reasonably be in FileHandle.h. But if/when we have a POSIX API, we would presumably want the real poll() from `<poll.h>` too. So maybe mbed_poll.h?

As the FileHandle version is inherently C++ maybe this could just be an overload of `poll()`?

It's `short` because that's the standard parameter for `poll()`.

I think the comments should be as explicit as possible that this is trying to be POSIX poll(). Not sure if the phrasing was copied from the POSIX spec?

This poll API doesn't quite line up with SIGIO - SIGIO doesn't provide definite answers. poll does. So it's a tougher requirement to implement - still not convinced its possible on all stacks.

Maybe the poll change callback could be turned into just SIGIO - non-definite. So apps could just use the SIGIO and be compatible with limited-functionality network stacks? And maybe apps could be written to tolerate false positives from poll()?

I am nervous about how we integrate this with sockets - at present this assumes reliable poll information, whereas I believe sockets won't always provide that. We potentially end up with a worse set of confusion (cf mbed client PAL, which provides a "select", but doesn't provide reliable answers on all platforms, but apps assume that it does).

The `_poll_change` is (or will be) required to make top-level poll() work. It covers "release the block in poll" as well as the application callback. (Although the block mechanism is not yet implemented). I'm thinking maybe it shouldn't be a member though. 

That member poll() function is potentially confusing, just from a naming point of view. Wondering if it should be called something else...